common/src/main/kotlin/io/homeassistant/companion/android/common/util/JsonUtil.kt

```diff
@@ -265,3 +267,160 @@ private fun toJsonElement(encoder: JsonEncoder, value: Any?): JsonElement {
         }
     }
 }
+
+/**
+ * Returns current element as JsonObject. If current element is not a JsonObject (ex/ JsonPrimitive) will swallow the exception and return null
+ */
+fun JsonElement.jsonObjectOrNull(): JsonObject? = runCatching {
+    this.jsonObject
+}.onFailure { ex ->
+    Timber.w("Failed to get current element as JsonObject $this, exception: $ex")
+}.getOrNull()
+
+/**
+ * Converts a map to a JsonObject
+ */
+
+fun Map<String, Any?>.toJsonObject(): JsonObject {
+    val content = this.mapValues { (_, value) ->
+        value.toJsonElementOrNull()
+    }
+    return JsonObject(content)
+}
+
+/**
+ * Parses a string to a JsonObject. If the string is empty this method will return null
```

Comment:
It will also return null if it is not a JsonObject. Take the time to go over the doc again 🙏🏻 public API like this needs to be well documented so we don't check the implementation while using it.